examples/Atmos/heldsuarez.jl

```diff
@@ -0,0 +1,253 @@
+                    FT              = %s
+                    polynomialorder = %d
+                    numelem_horz    = %d
+                    numelem_vert    = %d
+                    filterorder     = %d
+                    dt              = %.16e
+                    norm(Q₀)        = %.16e
+                    """ "$ArrayType" "$FT" polynomialorder numelem_horz numelem_vert filterorder dt eng0
+
+  # Set up the information callback
+  starttime = Ref(now())
+  cbinfo = EveryXWallTimeSeconds(60, mpicomm) do (s=false)
+    if s
+      starttime[] = now()
+    else
+      energy = norm(Q)
+      runtime = Dates.format(convert(DateTime, now() - starttime[]), dateformat"HH:MM:SS")
+      @info @sprintf """Update
+                        simtime = %.16e
+                        runtime = %s
+                        norm(Q) = %.16e
+                        """ gettime(lsrk) runtime energy
+    end
+  end
+  callbacks = (cbinfo, cbfilter)
+
+  if output_vtk
+    # create vtk dir
+    vtkdir = "vtk_heldsuarez" *
+      "_poly$(polynomialorder)_horz$(numelem_horz)_vert$(numelem_vert)" *
+      "_filter$(filterorder)_$(ArrayType)_$(FT)"
+    mkpath(vtkdir)
+
+    vtkstep = 0
+    # output initial step
+    do_output(mpicomm, vtkdir, vtkstep, dg, Q, model)
+
+    # setup the output callback
+    cbvtk = EveryXSimulationSteps(floor(outputtime / dt)) do
+      vtkstep += 1
+      Qe = init_ode_state(dg, gettime(lsrk))
+      do_output(mpicomm, vtkdir, vtkstep, dg, Q, model)
+    end
+    callbacks = (callbacks..., cbvtk)
+  end
+
+  solve!(Q, lsrk; timeend=timeend, callbacks=callbacks)
+
+  # final statistics
+  engf = norm(Q)
+  @info @sprintf """Finished
+  norm(Q)                 = %.16e
+  norm(Q) / norm(Q₀)      = %.16e
+  norm(Q) - norm(Q₀)      = %.16e
+  """ engf engf/eng0 engf-eng0
+end
+
+Base.@kwdef struct HeldSuarezSetup{FT}
+  p_ground::FT = MSLP
+  T_initial::FT = 255
+  domain_height::FT = 30e3
+end
+
+function (setup::HeldSuarezSetup)(state, aux, coords, t) 
+  # callable to set initial conditions
+  FT = eltype(state)
+
+  r = norm(coords, 2)
+  h = r - FT(planet_radius)
+
+  scale_height = R_d * setup.T_initial / grav
+  p = setup.p_ground * exp(-h / scale_height)
+
+  state.ρ = air_density(setup.T_initial, p)
+  state.ρu = SVector{3, FT}(0, 0, 0)
+  state.ρe = state.ρ * (internal_energy(setup.T_initial) + aux.orientation.Φ)
+  nothing
+end
+
+function held_suarez_forcing!(source, state, aux, t::Real)
+  FT = eltype(state)
+
+  ρ = state.ρ
+  ρu = state.ρu
+  ρe = state.ρe
+  coord = aux.coord
+  Φ = aux.orientation.Φ
+  e = ρe / ρ
+  u = ρu / ρ
+  e_int = e - u' * u / 2 - Φ
+  T = air_temperature(e_int)
+  # Held-Suarez constants
+  k_a = FT(1 / 40 / 86400)
+  k_f = FT(1 / 86400)
+  k_s = FT(1 / 4 / 86400)
+  ΔT_y = FT(60)
+  Δθ_z = FT(10)
+  T_equator = FT(315)
+  T_min = FT(200)
+  σ_b = FT(7 / 10)
+  r = norm(coord, 2)
+  @inbounds λ = atan(coord[2], coord[1])
+  @inbounds φ = asin(coord[3] / r)
+  h = r - FT(planet_radius)
```

Comment:
This way of determining altitude is fine for now. But it'd be good to make this more general at some point, so that we can have a non-spherical (geoid) planet, and then carry that more general altitude coordinate in the same way as the geopotential.